DRODLib/DbRooms.cpp

```diff
@@ -3776,8 +3779,21 @@ const
 		{
 			if (*pMonsters && (*pMonsters)->IsAlive())
 			{
-				if ((*pMonsters)->wType == wType)
-					return true;
+				if ((*pMonsters)->wType == wBaseType) {
+					if (wBaseType == wType)
+						return true;
+					else
+						switch(wType) {
+							case M_EYE_ACTIVE:
+							{
+								const CEvilEye* pEvilEye = DYN_CAST(CEvilEye*, CMonster*, (*pMonsters));
+								if (pEvilEye->IsAggressive())
+									return true;
+								continue;
```

Comment:
I think having a continue here is incorrect.

Note the pMonsters increment at the bottom of the loop.